core/src/proof.rs

```diff
@@ -6,12 +6,30 @@ use crate::trie::{
 
 use bitvec::prelude::*;
 
+/// Wrapper for a terminal node, it will store the LeafData if it is a leaf node,
+/// and just the KeyPath to that terminal if it is a terminator node
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum PathProofTerminal {
+    Leaf(LeafData),
+    Terminator(BitVec<u8, Msb0>),
+}
+
+impl PathProofTerminal {
+    /// Return the KeyPath of the Terminal Node
+    pub fn key_path(&self) -> &BitSlice<u8, Msb0> {
```

Comment:
nit: for a follow-up, I'd prefer if this were named `path` rather than `key_path`. we haven't actually defined this convention anywhere, but I treat "key path" as a full 32-byte path (based off of a _key_) and a "path" as the bit-path to any node in the trie.

Reply:
this makes perfectly sense! thanks!